app/controllers/settings/ext_cert_controller.rb

```diff
@@ -0,0 +1,260 @@
+      external_cert_velum_key:    key_cert_map_temp[:velum][:key][:key_string],
+      external_cert_kubeapi_cert: key_cert_map_temp[:kubeapi][:cert][:cert_string],
+      external_cert_kubeapi_key:  key_cert_map_temp[:kubeapi][:key][:key_string],
+      external_cert_dex_cert:     key_cert_map_temp[:dex][:cert][:cert_string],
+      external_cert_dex_key:      key_cert_map_temp[:dex][:key][:key_string]
+    }
+    @errors = Pillar.apply cert_map
+    if @errors.empty?
+      redirect_to settings_ext_cert_index_path,
+        notice: "External Certificate settings successfully saved."
+      return
+    # :nocov:
+    # An error here would require a failure in connection to velum->salt
+    # or a corruption in mapping of values in the salt pillar
+    else
+      set_instance_variables
+      render action: :index, status: :unprocessable_entity
+      # :nocov:
+    end
+  end
+
+  private
+
+  def set_instance_variables
+    @velum_cert = cert_parse(Pillar.value(pillar: :external_cert_velum_cert))
+    @velum_key = key_parse(Pillar.value(pillar: :external_cert_velum_key))
+    @kubeapi_cert = cert_parse(Pillar.value(pillar: :external_cert_kubeapi_cert))
+    @kubeapi_key = key_parse(Pillar.value(pillar: :external_cert_kubeapi_key))
+    @dex_cert = cert_parse(Pillar.value(pillar: :external_cert_dex_cert))
+    @dex_key = key_parse(Pillar.value(pillar: :external_cert_dex_key))
+  end
+
+  def get_val_from_form(param)
+    if params.key?(:external_certificate) && params[:external_certificate].key?(param)
+      params[:external_certificate][param].read.strip
+    else
+      ""
+    end
+  end
+
+  def key_cert_map
+    {
+      velum:   {
+        name: VELUM_NAME,
+        cert: {
+          cert_string:      get_val_from_form(:velum_cert),
+          pillar_model_key: :external_cert_velum_cert
+        },
+        key:  {
+          key_string:       get_val_from_form(:velum_key),
+          pillar_model_key: :external_cert_velum_key
+        }
+      },
+      kubeapi: {
+        name: KUBEAPI_NAME,
+        cert: {
+          cert_string:      get_val_from_form(:kubeapi_cert),
+          pillar_model_key: :external_cert_kubeapi_cert
+        },
+        key:  {
+          key_string:       get_val_from_form(:kubeapi_key),
+          pillar_model_key: :external_cert_kubeapi_key
+        }
+      },
+      dex:     {
+        name: DEX_NAME,
+        cert: {
+          cert_string:      get_val_from_form(:dex_cert),
+          pillar_model_key: :external_cert_dex_cert
+        },
+        key:  {
+          key_string:       get_val_from_form(:dex_key),
+          pillar_model_key: :external_cert_dex_key
+        }
+      }
+    }
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def upload_validate(key_cert_map)
+    # Do nothing if both cert/key are empty
+    if key_cert_map[:cert][:cert_string].empty? && key_cert_map[:key][:key_string].empty?
+      true
```

Comment:
An early return without return looks really terrible. I would advise to override rubocop on this as it looks nutty.

I can see why it is the case, because it is the last thing evaluated, but because the code below goes down so far it is not immediately obvious that is the last evaluated value.

At the very least a 'true # return true' is in order.

Reply:
Agree completely, but as you noted this is what rubocop wants.  I'll add the `# return true` comment.